### PR TITLE
Updated post message system

### PR DESCRIPTION
**Pull request by HarrisonHough**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,17 +1,77 @@
...
+import { WebView } from "react-native-webview";
+
+// Replace with your custom subdomain
+const subdomain = "demo";
+
+let isSubscribed = false;
+let count = 0;
+const correlationId = "a0bf9c2a-44d7-4882-8e72-4bc7ab73849f";
+
+function OnAvatarExported(json) {
```
Comment (ygrishajev) [inline App.js:11]:
```suggestion
function onAvatarExported(message) {
```

File: App.js [hunk 2]
```diff
@@ -1,17 +1,77 @@
...
+  alert(`Avatar Url = ${json.data?.url}`);
+}
+
+const View = () => {
+  const webview = useRef();
+  const subscribe = () => {
+    if (isSubscribed) {
```
Comment (ygrishajev) [inline App.js:18]: I'd suggest using `useEffect` to subscribe. Would be no need to use a boolean to track this.

File: App.js [hunk 3]
```diff
@@ -1,17 +1,77 @@
...
+    webview.current.postMessage(
+      JSON.stringify({
+        target: "readyplayerme",
+        type: "subscribe",
+        eventName: "v1.avatar.exported",
+      })
+    );
+  };
+
+  const process = (data) => {
```
Comment (ygrishajev) [inline App.js:32]: should be wrapped into `useCallback`

**Review by ygrishajev — APPROVED** [review]
I realize this is an example but it's gonna be public, so I'd suggest to follow some best practices as you'd do it for any production code.
If you believe this is not the case for some reason and a working example is enough just ignore my comments.

File: App.js (hunk 1 above)
Comment (HarrisonHough) [reply App.js:11]: updated

File: App.js (hunk 2 above)
Comment (HarrisonHough) [reply App.js:19]: hmm I tried this but it didn't seem to work? 

```js
const View = () => {
  const webview = useRef();

  useEffect(() => {
    webview.current.postMessage(
      JSON.stringify({
        target: "readyplayerme",
        type: "subscribe",
        eventName: "v1.avatar.exported",
      })
    );
  });
  ....
```

**Pull request closed without merging**